### ISSUE TITLE
Need to get pod cidr and DNS address to worker kubelet

### ISSUE THREAD

**Issue by jnummelin** [k0sproject/k0s#155]
In #152 we fixed most of the networking related misconfigs. To keep that PR somewhat manageable we hardcoded the pod CIDR and DNS addresses into kubelet config.

We need to somehow be able to transfer those details, and possibly other, from control plane to worker (kubelet).

## Use configmaps on API

We could use similar way as kubeadm does and store a generic `kubelet-1.19` configmap on the API. When bootstrapping the worker, mke could retrieve that and use for kubelet config. Once we have #54 implemented, we'd need some way to "merge" the common config with the local config.

#### Possible issues (or implementation pain)

After the initial bootstrapping, can we use the real kubelets auth config to access said configmaps? The auth detail embedded into the "token" are very probably expired already and thus not usable.

## Use "token" for smuggling

We could "extend" the token we use for worker join to have these attributes e.g. in the `preferences` section. When bootstrapping the worker, we can dump these into kubelet config. One possible problem is how to use the attributes after the initial bootstrapping as we do not need the token anymore, and it might be expired already.

#### Possible issues (or implementation pain)

After the initial bootstrapping, we should not rely on the "token" to be valid anymore. But could we still use it to get these bits of information and merge those into real kubelet config? 

## Use dynamic kubelet config

There's way to create kubelet config as a config map and assign it to a particular node in `node.spec...`. https://kubernetes.io/docs/tasks/administer-cluster/reconfigure-kubelet/

To make the assignment work we'd need to have something running in mke control plane that ensures there's a config assigned into each node when node registers. An operator of sorts. We'd need a way to be able to assign maybe different profiles into the nodes based on user configuration. Not all nodes are similar and might require very different kubelet configs.

#### Possible issues (or implementation pain)

Lot of work. :) How would a node indicate which "profile" config it needs? Kubelet CANNOT anymore push labels itself.

# TL;DR;

For short term, IMO, using the token to smuggle these bits might be the easiest choice. For longer run we'd probably want to take a deeper look if we can make the profile based kubelet config assignment a reality.
